### PR TITLE
PFMENG-203 - Fix vault jwt auth path

### PR DESCRIPTION
**Pull request by smoneyan**
main

### CODE REVIEW

File: hooks/terraform-pre-apply [hunk 1]
```diff
@@ -105,7 +105,7 @@ if [ -z ${TFC_VAULT_APPLY_ROLE+x} ] && [ -z ${TFC_VAULT_RUN_ROLE+x} ]; then
       "jwt": $JWT, 
       "role": $ROLE }' > ~/.vault-payload.json
 
-    VAULT_TOKEN=$(curl --silent --show-error "$VAULT_ADDR/v1/auth/jwt/login" \
+    VAULT_TOKEN=$(curl --silent --show-error "$VAULT_ADDR/v1/auth/tfc/login" \
```
Comment (lawliet89) [inline hooks/terraform-pre-apply:108]: Let's add another environment variable that defaults to `tfc`.

Maybe `VAULT_AUTH_PATH`

Comment (smoneyan) [reply hooks/terraform-pre-apply]: done. 

https://github.com/SPHTech-Platform/terraform-cloud-configuration/pull/413

File: hooks/terraform-pre-apply [hunk 2]
```diff
@@ -105,7 +105,7 @@ if [ -z ${TFC_VAULT_APPLY_ROLE+x} ] && [ -z ${TFC_VAULT_RUN_ROLE+x} ]; then
       "jwt": $JWT, 
       "role": $ROLE }' > ~/.vault-payload.json
 
-    VAULT_TOKEN=$(curl --silent --show-error "$VAULT_ADDR/v1/auth/jwt/login" \
+    VAULT_TOKEN=$(curl --silent --show-error "$VAULT_ADDR/v1/auth/$VAULT_AUTH_PATH/login" \
```
Comment (lawliet89) [inline hooks/terraform-pre-apply:108]: Add some code at the top of the script to default to `tfc` if value is not set.

File: test/_helpers.bash [hunk 3]
```diff
@@ -9,6 +9,7 @@ setup() {
   export TFC_GCP_WORKLOAD_PROVIDER_ID="tfc"
   export VAULT_ADDR="https://main.vault.0cbdb520-5871-4f8a-b02b-81e27b14da3f.aws.hashicorp.cloud:8200/"
   export VAULT_NAMESPACE="test"
+  export VAULT_AUTH_PATH="tfc"
```
Comment (lawliet89) [inline test/_helpers.bash:12]: I think you can remove this to make sure the default setting works.

File: hooks/terraform-pre-plan [hunk 4]
```diff
@@ -98,14 +98,15 @@ if [ -z ${TFC_VAULT_PLAN_ROLE+x} ] && [ -z ${TFC_VAULT_RUN_ROLE+x} ]; then
   else
     set +u
     VAULT_ROLE="${TFC_VAULT_PLAN_ROLE:-$TFC_VAULT_RUN_ROLE}"
+    VAULT_PATH="${VAULT_AUTH_PATH:-tfc}"
```
Comment (lawliet89) [inline hooks/terraform-pre-plan:101]: Put this outside of `set +u`.

File: test/_helpers.bash (hunk 3 above)
Comment (smoneyan) [reply test/_helpers.bash]: yeah.. correct. removed it.

**Review by lawliet89 — APPROVED** [review]

**Pull request merged**